ecosystem/typescript/sdk/src/aptos_types/transaction.ts

```diff
@@ -371,25 +371,61 @@ export abstract class RawTransactionWithData {
     switch (index) {
       case 0:
         return MultiAgentRawTransaction.load(deserializer);
+      case 1:
+        return MultiAgentRawTransaction.load_fee_payer(deserializer);
       default:
         throw new Error(`Unknown variant index for RawTransactionWithData: ${index}`);
     }
   }
 }
 
 export class MultiAgentRawTransaction extends RawTransactionWithData {
+  private fee_payer_address?: AccountAddress;
+
   constructor(
     public readonly raw_txn: RawTransaction,
     public readonly secondary_signer_addresses: Seq<AccountAddress>,
   ) {
     super();
   }
 
+  set_fee_payer_address(fee_payer_address: AccountAddress): void {
+    this.fee_payer_address = fee_payer_address;
+  }
+
+  get_authenticator(
+    sender: AccountAuthenticator,
+    secondary_signers: Seq<AccountAuthenticator>,
+    fee_payer?: AccountAuthenticator,
+  ): TransactionAuthenticatorMultiAgent {
+    if (secondary_signers.length !== this.secondary_signer_addresses.length) {
+      throw new Error("Number of secondary signers does not match.");
+    }
+    let fee_pay;
+    if (this.fee_payer_address) {
+      if (!fee_payer) {
```

Comment:
a bit confused here, at this point, `fee_payer` is always undefined b/c of line 404?